### PR TITLE
RUM-7091: Add TextField semantics mapper for Session Replay Compose

### PR DESCRIPTION
**Pull request by ambushwork**
### What does this PR do?

Add TextField semantics mapper for Session Replay Compose, with this PR, following compose components can be supported:
* TextField
* OutlineTextField

Known issue of this mapper:
* TextField hint text color might not be correct,
* OutlineTextField can have the correct outline

### Motivation

RUM-7091

### Demo

| Sample App |.  Session Replay |
| --- | --- |
| Screenshot_20241121_095512| ezgif-7-7d88bddb00 |

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/RootSemanticsNodeMapper.kt [hunk 1]
```diff
@@ -95,7 +99,13 @@ internal class RootSemanticsNodeMapper(
         semanticsNode: SemanticsNode
     ): SemanticsNodeMapper {
         val role = semanticsNode.config.getOrNull(SemanticsProperties.Role)
-        return semanticsNodeMapper[role] ?: if (isTextNode(semanticsNode)) {
+        val mapper = semanticsNodeMapper[role]
+        if (mapper != null) {
+            return mapper
+        }
+        return if (isTextFieldNode(semanticsNode)) {
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/RootSemanticsNodeMapper.kt:106]:
```suggestion
        return if (mapper != null) {
           mapper
        } else if (isTextFieldNode(semanticsNode)) {
```

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/RootSemanticsNodeMapper.kt [hunk 2]
```diff
@@ -107,6 +117,10 @@ internal class RootSemanticsNodeMapper(
         return semanticsNode.config.getOrNull(SemanticsProperties.Text)?.isNotEmpty() == true
     }
 
+    private fun isTextFieldNode(semanticsNode: SemanticsNode): Boolean {
+        return semanticsNode.config.getOrNull(SemanticsActions.SetText) != null
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/RootSemanticsNodeMapper.kt:121]: minor:

```suggestion
        return semanticsNode.config.contains(SemanticsActions.SetText)
```

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/TextFieldSemanticsNodeMapper.kt [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                    )
+                )
+            )
+        }
+    }
+
+    private fun resolveEditText(semanticsConfiguration: SemanticsConfiguration): String? {
+        return semanticsConfiguration.getOrNull(SemanticsProperties.EditableText)?.let {
+            return resolveAnnotatedString(it)
+        }
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/TextFieldSemanticsNodeMapper.kt:103]:
```suggestion
        return semanticsConfiguration.getOrNull(SemanticsProperties.EditableText)?.let {
           resolveAnnotatedString(it)
        }
```

File: features/dd-sdk-android-session-replay-compose/src/test/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/TextFieldSemanticsNodeMapperTest.kt [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        mockColorStringFormatter(fakeBackgroundColor, fakeBackgroundColorHexString)
+        mockColorStringFormatter(fakeTextLayoutInfo.color.toLong(), fakeTextColorHexString)
+        whenever(mockDensity.density) doReturn fakeDensity
+        testedTextFieldSemanticsNodeMapper = TextFieldSemanticsNodeMapper(
+            colorStringFormatter = mockColorStringFormatter,
+            semanticsUtils = mockSemanticsUtils
+        )
+    }
+
+    private fun mockSemanticsNode(): SemanticsNode {
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay-compose/src/test/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/TextFieldSemanticsNodeMapperTest.kt:96]: to be moved to the EOF

**Review by 0xnm — APPROVED** [review]
few suggestions, but overall lgtm

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/RootSemanticsNodeMapper.kt (hunk 1 above)
Comment (ambushwork) [reply features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/RootSemanticsNodeMapper.kt]: updated

**Pull request merged**